src/Common/src/CoreLib/System/WinRTFolderPaths.cs

```diff
@@ -31,109 +31,74 @@ public static string GetFolderPath(SpecialFolder folder, SpecialFolderOption opt
             }
         }
 
-        private static string GetFolderPathCoreCurrent(SpecialFolder folder)
-        {
+        private static string GetFolderPathCoreCurrent(SpecialFolder folder) =>
             // While all of these give back real paths, most of them are not accessible
             // from an appcontainer currently (they will give access denied)
-            switch (folder)
+            folder switch
```

Comment:
Just want to confirm. This doesn't regress codegen/etc, correct?

Reply:
Good question.  I'd checked a couple patterns but in checking some others, it appears that the JIT is doing a good job of producing the same asm, but the generated IL ends up being a bit bigger in some cases.  For example:
https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDYB8ACAmARgFgAoHAZgAJ9KBhSgb1MpeqrQDsZKAlOGAK4IOBABSdunACZwAHgEpGzViqgB3NDADGAC0riOMhcpUsmJU5cpbocSgAYQ1AOyUALHgDcJqyxtQ7AiccVwIADi8fX387PGDXAE4ANm8LX1YYynJ4ygI8cjcAVlT01hkAMwgBDFQXBxKrAF8fZrSWUgBICkoJXn4hDjwDSUM5RXNfXoQ4KGqYBst1TV19aTGo1gnS61sHJ2nZmsoAXncvSmBpiABrBfTMoMoDuZPciM8Lq9uNy0y4p5mL1OyQ+lzgNzu0V22QBh24pzyBWKn3B3za6QqVRq+0BR1O9lBX0hrFavhCsLmd1JrB83V6fEEwnIwx6owUJwAfD9WUZKEttDpuVtSvZOWckNyVAQxeE8BL0b48GLkvLtlkZfkiqrtgB9MX2bWWRqpRpAA==
In that example, Return1 has smaller IL than Return2/Return3, which are the same, but they all produce identical asm.

@gafter, @jaredpar, is this something the C# compiler can fix?

Reply:
I looked at assembly sizes across corefx after this change.  Looking at all of the built assemblies (System.* and Microsoft.*), size increased from 31,289,160 bytes to 31,303,496 bytes, for a 0.04% increase (14K).  That's small enough that I think it's worth taking the cleanup / modernizing the source, in particular since the asm for the examples I looked at ends up being the same.  However, it would still be nice if the compiler could improve the generated IL; @jaredpar pinged me offline to say it's something the compiler can look at improving.